libs/wingsdk/src/shared-aws/aws-function.ts

```diff
@@ -0,0 +1,54 @@
+import { PolicyStatement } from "./types";
+import { IInflightHost } from "../std";
+import { Function as AwsCdkFunction } from "../target-awscdk";
+import { Function as TfAwsFunction } from "../target-tf-aws";
+
+/**
+ * A helper interface for working with AWS functions.
+ */
+export interface IAwsFunction {
+  /**
+   * Add an environment variable to the function.
+   */
+  addEnvironment(key: string, value: string): void;
+  /**
+   * Add a policy statement to the function's IAM role.
+   */
+  addIamPolicy(policy: PolicyStatement): void;
+}
+
+/**
+ * A helper class for working with AWS functions.
+ */
+export class AwsFunction {
+  /**
+   * If the inflight host is an AWS function, return a helper interface for
+   * working with it.
+   * @param host The inflight host.
+   */
+  public static from(host: IInflightHost): IAwsFunction | undefined {
+    if (host instanceof TfAwsFunction) {
```

Comment:
Why not just return `host`?

Make TfAwsFunction and AwsCdkFunction implement IAwsFunction and just return them directly. No?

Reply:
Yeah that would be a lot nicer - let me give it a try. I initially jumped over this solution because I thought I might run into some kind of circular import issue between namespaces, but I suspect this is just a case of me overthinking things :)

Reply:
Done